aws/ec2metadata/api.go

```diff
@@ -8,14 +8,15 @@ import (
 	"strings"
 	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/internal/sdkuri"
 )
 
 // getToken uses the duration to return a token for EC2 metadata service,
 // or an error if the request failed.
-func (c *EC2Metadata) getToken(duration time.Duration) (tokenOutput, error) {
+func (c *EC2Metadata) getToken(ctx aws.Context, duration time.Duration) (tokenOutput, error) {
```

Comment:
Some of the unexported methods are taking an `aws.Context` and other `credentials.Context`. What do you think about standardizing only only using `credentials.Context` for the exported `RetrieveWithContext`, and `aws.Context` for everything else? Mainly just for consistency with the SDK's internal. `credentials.Context` only exists as a redef of `aws.Context`, because `aws` imports `credentials` package.